### PR TITLE
Use emtpy string for empty comments

### PR DESCRIPTION
**Pull request by twennemann**
- Changed the initialization of comments from "none" to and empty string "".
- revised the tests.
- comments which are not of type string will no longer be casted. A TypeError will be raised instead.

### CODE REVIEW

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks, looks good to me - only the documentation fails on circle and needs fixing. Do you want help with that?

File: examples/pyfar_demo.ipynb [hunk 1]
```diff
@@ -749,7 +749,7 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 3.10.4 ('pyfar')",
```
Comment (sikersten) [inline examples/pyfar_demo.ipynb:752]: please remove these not related changes from the PR :)

File: tests/test_dsp_average.py [hunk 2]
```diff
@@ -7,15 +7,15 @@
...
      'linear', [2.5, 3.5, 4.5]],
     [pf.signals.impulse(128, [0, 2], [1, 3]),
      'magnitude_zerophase', np.zeros(65)+2],
     [pf.signals.impulse(128, [0, 2], [1, 3]),
      'magnitude_phase', pf.signals.impulse(128, 1, 2).freq[0]],
-    [pf.FrequencyData([[1, 2, 3], [4, 5, 6]], [1, 2, 3], 44100),
+    [pf.FrequencyData([[1, 2, 3], [4, 5, 6]], [1, 2, 3]),
      'power', np.sqrt([(1+16)/2, (4+25)/2, (9+36)/2])],
-    [pf.FrequencyData([[0.01, 0.1, ], [1, 10]], [1, 2], 44100),
+    [pf.FrequencyData([[0.01, 0.1, ], [1, 10]], [1, 2]),
```
Comment (sikersten) [inline tests/test_dsp_average.py:18]: It looks like these changes are not related to the PR? Please move to a separate one :)

**Review by sikersten — CHANGES_REQUESTED** [review]
Look good thanks! Please remove code that is not related to this PR, otherwise approved :)

File: pyfar/classes/audio.py [hunk 3]
```diff
@@ -201,8 +204,10 @@ def __init__(self, data, times, comment=None):
         times : array, double
             Times in seconds at which the data is sampled. The number of times
             must match the `size` of the last dimension of `data`.
-        comment : str, optional
-            A comment related to `data`. The default is ``'none'``.
+        comment : str
+            A comment related to `data`. The default is ``''``, which
+            initializes an empty string.
+.
```
Comment (sikersten) [inline pyfar/classes/audio.py:210]: This should fix the documentation build (empty line instead of full stop)
```suggestion

```

**Comment by sikersten** [comment]
Closes issue #359

**Review by f-brinkmann — APPROVED** [review]

**Review by sikersten — APPROVED** [review]

**Pull request merged**